frontend/src/lib/user.ts

```diff
@@ -216,11 +216,18 @@ function projectsStringToProjects(projectsString: string | undefined): AuthUserP
         role = ProjectRole.Editor;
         break;
     }
-    projects.push(...pString.split('|').map(id => ({projectId: id, role})));
+    //substring to remove the first character which is the role code plus the colon
```

Comment:
Oh wow, has this code been buggy this whole time, so that the project-roles list never had correct UUIDs and so we would never have matched them correctly? I don't know why we haven't noticed until now.

Reply:
yeah it looks like it has. Not sure how we didn't catch it. It was only client side so maybe that's why?